src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,271 @@
+pub struct TaskInput {
+	job_id: i32,
+	input: CheckEmailInput,
+}
+
+/// Endpoint request body.
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct CreateBulkRequestBody {
+	input_type: String,
+	input: Vec<String>,
+	proxy: Option<HashMap<String, String>>,
+	hello_name: Option<String>,
+	from_email: Option<String>,
+	smtp_port: Option<usize>,
+}
+
+pub struct CreateBulkRequestBodyIterator {
+	body: CreateBulkRequestBody,
+	index: usize,
+	batch_size: usize,
+}
+
+impl IntoIterator for CreateBulkRequestBody {
+	type Item = CreateBulkRequestBody;
+	type IntoIter = CreateBulkRequestBodyIterator;
+
+	fn into_iter(self) -> Self::IntoIter {
+		CreateBulkRequestBodyIterator {
+			body: self,
+			index: 0,
+			batch_size: EMAIL_TASK_BATCH_SIZE,
+		}
+	}
+}
+
+impl Iterator for CreateBulkRequestBodyIterator {
+	type Item = CreateBulkRequestBody;
+
+	fn next(&mut self) -> Option<Self::Item> {
+		if self.index <= self.body.input.len() {
+			let item = Some(CreateBulkRequestBody {
+				input_type: self.body.input_type.clone(),
+				input: self.body.input[self.index..self.index + self.batch_size].to_vec(),
+				proxy: self.body.proxy.clone(),
+				hello_name: self.body.hello_name.clone(),
+				from_email: self.body.from_email.clone(),
+				smtp_port: self.body.smtp_port.clone(),
+			});
+
+			self.index = self.index + self.batch_size;
+
+			item
+		} else {
+			None
+		}
+	}
+}
+
+impl Into<CheckEmailInput> for CreateBulkRequestBody {
+	fn into(self) -> CheckEmailInput {
+		let mut input = CheckEmailInput::new(self.input);
+
+		if let Some(name) = self.hello_name {
+			input.set_hello_name(name);
+		}
+
+		if let Some(email) = self.from_email {
+			input.set_from_email(email);
+		}
+
+		if let Some(port) = self.smtp_port {
+			input.set_smtp_port(port as u16);
+		}
+
+		if let Some(proxy) = self.proxy {
+			input.set_proxy(CheckEmailInputProxy {
+				host: proxy.get("host").unwrap().clone(),
+				port: proxy.get("port").unwrap().parse().unwrap(),
+			});
+		}
+
+		input
+	}
+}
+
+/// Endpoint response body.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CreateBulkResponseBody {
+	job_id: i32,
+}
+
+/// Endpoint error response
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ErrorResponseBody {
+	error: String,
+}
+
+// Arguments to the `#[job]` attribute allow setting default job options.
+/// This task tries to verify the given email and inserts the results
+/// into the email verification db table
+#[job(channel_name = "foo")]
+pub async fn email_verification_task(
+	mut current_job: CurrentJob,
+	// Additional arguments are optional, but can be used to access context
+	// provided via [`JobRegistry::set_context`].
+) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+	let task_input: TaskInput = current_job.json()?.unwrap();
+
+	// Retry each future twice, to avoid grey-listing.
+	if let Ok((response, _)) = retry(&task_input.input, RetryOption::Direct, 2).await {
```

Comment:
What do we do in case of error?

One simple solution would be to add a new column `ema_vrfy_rec` called `error`. And exactly one column of `(result, error)` should be populated in the table.

Reply:
Storing an enum in the db is slightly non-trivial. In the sense that the whole field will be a JSONB and the json itself will have a structure like this - `{ type: "result" (or "error"), "data": (actual data) }`.

This is what serde offers. This will require changing the aggregate query to check the `type` field to make sure it's a result before trying to check it's verification status.

Also error's might be pretty large. How do you intend you use the error information, is logging it not enough?

Reply:
yeah ok, let's just store the `response` JSON, like you're currently doing.